src/core/io/IE_fbt.cpp

```diff
@@ -35,44 +35,29 @@ USE_STRING_ID(STRING);
 
 using namespace forte::core::io;
 
-DEFINE_FIRMWARE_FB(FORTE_IE, STRID(IE))
+namespace {
+  const auto cEventInputNames = std::array{STRID(INIT), STRID(REQ)};
+  const auto cEventInputTypeIds = std::array{STRID(EInit), STRID(Event)};
+  const auto cEventOutputNames = std::array{STRID(INITO), STRID(CNF), STRID(IND)};
+  const auto cEventOutputTypeIds = std::array{STRID(EInit), STRID(Event), STRID(Event)};
+  const auto cDataInputNames = std::array{STRID(QI), STRID(PARAMS)};
+  const auto cDataOutputNames = std::array{STRID(QO), STRID(STATUS)};
+
+  const SFBInterfaceSpec cFBInterfaceSpec = {.mEINames = cEventInputNames,
+                                             .mEITypeNames = cEventInputTypeIds,
+                                             .mEONames = cEventOutputNames,
+                                             .mEOTypeNames = cEventOutputTypeIds,
+                                             .mDINames = cDataInputNames,
+                                             .mDONames = cDataOutputNames,
+                                             .mDIONames = {},
+                                             .mSocketNames = {},
+                                             .mPlugNames = {}};
```

Comment:
I suggest adding a comma after the last initializer, which tells `clang-format` to use block formatting:
```suggestion
  const SFBInterfaceSpec cFBInterfaceSpec = {
      .mEINames = cEventInputNames,
      .mEITypeNames = cEventInputTypeIds,
      .mEONames = cEventOutputNames,
      .mEOTypeNames = cEventOutputTypeIds,
      .mDINames = cDataInputNames,
      .mDONames = cDataOutputNames,
      .mDIONames = {},
      .mSocketNames = {},
      .mPlugNames = {},
  };
```

Reply:
didn't know that. Like it much more